src/azure-cli/azure/cli/command_modules/sqlvm/custom.py

```diff
@@ -475,57 +671,436 @@ def set_assessment_properties(cmd, instance, enable_assessment, enable_assessmen
     # 2. Custom log definition on workspace
     if enable_assessment:
         workspace_id = None
+        curr_subscription = get_subscription_id(cmd.cli_ctx)
+        # Raise error if workspace arguments not provided by user
+        if workspace_name is None or workspace_rg is None:
+            raise RequiredArgumentMissingError(
+                'Assessment requires a Log Analytics workspace and Log Analytics extension on VM - '
+                'workspace name and workspace resource group must be specified to deploy pre-requisites.')
+        if agent_rg is None:
+            raise RequiredArgumentMissingError(
+                'Assessment requires a Resource Group to deploy the AMA Agent resources- '
+                'Use the --agent-rg paramater to specify a resource group in this subscription. '
+                'It is recommended to reuse this resource group for other Assessment deployments using this Log Analytics workspace')
```

Comment:
Seems like `agent_rg` is a newly added required argument, right?

Reply:
Just noticed your PR description. Can you add a default value for `agent_rg` with warning message? We don't allow breaking change in this sprint. Our next breaking change window is Oct 2023

Reply:
fixing CI and adding non breaking change default value for agent_rg. Opening new PR as this one was closed in merge attempt